src/main/java/bennett/earthquake/EarthquakeFrame.java

```diff
@@ -0,0 +1,111 @@
+package bennett.earthquake;
+
+import bennett.earthquake.json.Feature;
+import bennett.earthquake.json.FeatureCollection;
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class EarthquakeFrame extends JFrame {
+
+    private JList<String> jlist = new JList<>();
+    private ButtonGroup buttonGroup = new ButtonGroup();
+    private JRadioButton oneHourButton = new JRadioButton("One Hour");
+    private JRadioButton thirtyDaysButton = new JRadioButton("Thirty Days");
+
+    private FeatureCollection featureCollection;
+
+    public EarthquakeFrame() {
+
+        setTitle("EarthquakeFrame");
+        setSize(300, 600);
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        setLayout(new BorderLayout());
+
+        JPanel topPanel = new JPanel(new FlowLayout());
+        topPanel.add(oneHourButton);
+        topPanel.add(thirtyDaysButton);
+
+        buttonGroup.add(oneHourButton);
+        buttonGroup.add(thirtyDaysButton);
+
+        add(topPanel, BorderLayout.NORTH);
+        add(jlist, BorderLayout.CENTER);
+
+        oneHourButton.setSelected(true);
+
+        ActionListener radioButtonListener = new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (oneHourButton.isSelected()) {
+                    fetchData(new EarthquakeServiceFactory().getService().oneHour());
```

Comment:
Don't create the service more than once

Reply:
but dont i need to do it for each one, the 1 hour and 30 days?